File: python/resistor-color-trio/resistor_color_trio.py
bands = (
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
)


def label(colors):
    ohms = bands.index(colors[0]) * 10
    ohms += bands.index(colors[1])
    ohms *= 10 ** bands.index(colors[2])

    return f"{ohms} ohms" if ohms < 1000 else f"{ohms // 1000} kiloohms"

File: lua/resistor-color-trio/resistor-color-trio.lua
return {
  decode = function(c1, c2, c3)
    local bands = {
      black = 0,
      brown = 1,
      red = 2,
      orange = 3,
      yellow = 4,
      green = 5,
      blue = 6,
      violet = 7,
      grey = 8,
      white = 9,
    }

    local value = (bands[c1] * 10 + bands[c2]) * (10 ^ bands[c3])
    local unit = 'ohms'

    if value >= 1000 then
      value = value / 1000
      unit = 'kiloohms'
    end

    return value, unit
  end
}
